### PR TITLE
Extend functionality of LK FUGA Wiser Switch devices

### PR DESCRIPTION
**Pull request by hanskroner**
Add DDFs describing button events and bindings for LK FUGA Wiser Double Relay and Dimmer Switches.

### CODE REVIEW

**Comment by hanskroner** [comment]
LK FUGA Double Relay Switches unfortunately do not send commands when pressing and holding buttons, so `_HOLD` and `_HOLD_RELEASED` events don’t seem to be possible with them.

LK FUGA Dimmer Switches unfortunately do not send commands when pressing and holding the two dedicate scene buttons, so `_HOLD` and `_HOLD_RELEASED` events don’t seem to be possible with them - but they do work for the dim up and dim down buttons.

**Comment by Smanar** [comment]
It 's possible to add battery switch without code edition (other than tuya ofc) ?

**Comment by hanskroner** [comment]
> It 's possible to add battery switch without code edition (other than tuya ofc) ?

I'm not sure. I'm by no means an expert, and the documentation on DDFs is pretty thin. The Wiser devices "behave" differently, Zigbee-wise - I guess their firmware was developed by different contractors. I added bits to `button_map.json` in a previous PR to make the Battery Switch's generate `_HOLD` and `_HOLD_RELEASED` events, and there's some idiosyncrasies that I couldn't figure out how to handle cleanly with a JS parser - so I went the route of adding the button maps.

Both devices covered in this PR are mains-powered though, and behave a bit differently than the Battery Switch.

**Comment by Smanar** [comment]
The button map is used by battery switch.
Your device is a router ?
Wired switches are bulbs (most of them) so they are working like them, and not use ZHAswitch.

**Comment by hanskroner** [comment]
LK FUGA Wiser is a product line with many different products. I think this is where the confusion is stemming from.

> The button map is used by battery switch.

The LK FUGA Wiser 4 Button Battery Switch uses a battery and has 4 buttons - as the name cleverly implies. I extended the functionality of that button map in a previous PR. This PR deals with two different products. They also have 4 buttons, but are mains powered and in addition to the physical switches have local loads attached to them - one has two relays and the other has a dimmer. This PR also adds button maps for these two devices' buttons.

> Your device is a router ?

Both devices covered by this PR are mains-powered and Zigbee routers. The product pages for the products are unfortunately in Danish, but their user guides do include an English section if you scroll a bit down. Maybe they can help illustrate what these products are:
Double Relay Switch: https://www1.lk.dk/katalog/vejledning/GDE7031901-03.pdf
Dimmer Switch: https://www1.lk.dk/katalog/vejledning/GDE7032001-03.pdf

> Wired switches are bulbs (most of them) so they are working like them, and not use ZHAswitch.

It would be perhaps easier to show a screen shot of the device's endpoints, but I'll try my best to explain it. In the case of the DImmer Switch, the device has 5 endpoints. Each of its physical buttons is represented by an endpoint which has Client clusters for things like On/Off, Level Control, and Scenes. Each of these endpoints behaves like a switch. There's 4 physical buttons, so there's 4 endpoints that each represent a button. The remaining endpoint represents the devices "internal" dimmer and has Server clusters for things like On/Off, Level Control, Ballast Configuration, and Scenes. This endpoint behaves like a dimmable light. In DDF terms (I think) the Double Relay Switch has 3 subdevices: 2 `$TYPE_ON_OFF_LIGHT` and 1 `$TYPE_SWITCH` while the Dimmer Switch has 2 subdevices: 1 `$TYPE_ON_OFF_LIGHT` and 1 `$TYPE_SWITCH`.

> So can be the issue, the device just send as return the state or the light level. Not a button state.

There's no issue because everything is on its own endpoint. Additionally, buttons are stateless - they're just push buttons. In the case of the Double Relay, because each physical relay is represented by a separate endpoint - they can be toggled independent of each other.

Both devices are working fine with what this PR contains now but I have left it as a draft because I would like to use a little more time to think about the proper way to expose the Dimmer Switch's Ballast Configuration as a `/config` key over the REST API.

**Comment by hanskroner** [comment]
After looking into the Ballast Configuration ZCL some more, I think that functionality would be better handled in a separate PR. I'm going to mark this one as "Ready for review" and work on the extra functionality for the dimmer separately.

**Comment by jlaur** [comment]
@hanskroner - do you know if it is possible to control the LED's on the four switches? It would be really cool being able to configure the rightmost LED's on the double relay to turn red when lights are on, for example outside a bathroom to indicate that it's occupied. I also had a look at the four Home Automation Dimmer Switches recently, but didn't manage to figure out what they are, or how to use them.

File: devices/wiser/fuga_double_relay_switch.json [hunk 1]
```diff
@@ -1,211 +1,205 @@
...
-      ]
-    }
-  ]
-}
\ No newline at end of file
+{
+  "schema": "devcap1.schema.json",
+  "manufacturername": "Schneider Electric",
+  "modelid": "LK Switch",
+  "product": "LK Fuga Doube Relay Switch",
```
Comment (jlaur) [inline devices/wiser/fuga_double_relay_switch.json:5]: Perhaps this typo could be fixed?
```suggestion
  "product": "LK Fuga Double Relay Switch",
```

**Comment by hanskroner** [comment]
> @hanskroner - do you know if it is possible to control the LED's on the four switches?

It doesn't look like it's possible. The LEDs follow the state of the relays/dimmer. I don't have a Wiser gateway, so I can't really say whether that's a supported feature or not.

**Comment by jlaur** [comment]
> The LEDs follow the state of the relays/dimmer.

What do you mean by that? The LED's on my relay are white for a brief moment when pressing the corresponding buttons, otherwise they are turned off.

> I don't have a Wiser gateway, so I can't really say whether that's a supported feature or not.

I don't have one either. I've seen in videos that they can be green/red when turned on/off, but also that it's configurable using the Wiser app.

**Comment by EjvindHald** [comment]
I am really looking forward to this. @manup - hope you have time for review, so it can be merged.

**Comment by hanskroner** [comment]
> What do you mean by that? The LED's on my relay are white for a brief moment when pressing the corresponding buttons, otherwise they are turned off.

The red/green behavior in the video is what I see on mine. 

> [...] but also that it's configurable using the Wiser app. For example: https://www.youtube.com/watch?v=k8OTKfjcKcw&t=72s

The switches support a manufacturer proprietary cluster, and I would guess that's used for configuring things like this behavior. However, without documentation on that cluster, you'd need someone with a Wiser gateway to capture the Zigbee traffic and understand how that cluster works.

**Comment by jlaur** [comment]
> The red/green behavior in the video is what I see on mine.

Hmm, that's strange. Did it work like this since you connected it to power for the first time, and without any configuration? I'm wondering if this depends on firmware version, and also whether that configuration would get lost if configured through a Wiser gateway, then re-paired with ConBee II.

**Comment by manup** [comment]
@hanskroner thanks the PR looks good to me. I agree that extended features of the switches can be added in later PRs.

So I think the PR can be merged for the release?

**Comment by hanskroner** [comment]
Great! Thanks @manup - ok from me to get it merged!

**Pull request merged**